packages/ra-core/src/dataProvider/useGetList.ts

```diff
@@ -87,12 +89,21 @@ export const useGetList = <RecordType extends RaRecord = any>(
             ...options,
             onSuccess: value => {
                 // optimistically populate the getOne cache
-                value?.data?.forEach(record => {
-                    queryClient.setQueryData(
-                        [resource, 'getOne', { id: String(record.id), meta }],
-                        oldRecord => oldRecord ?? record
-                    );
-                });
+                if (
+                    value?.data &&
+                    value?.data?.length <= MAX_DATA_LENGTH_TO_CACHE
```

Comment:
the `?` here and line 96 is normally not necessary.

Reply:
Fixed